tests/Integration/Classes/ObjectModelTest.php

```diff
@@ -0,0 +1,678 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Classes;
+
+use Configuration;
+use Db;
+use Language;
+use PHPUnit\Framework\TestCase;
+use Shop;
+use Tests\Resources\classes\TestableObjectModel;
+
+class ObjectModelTest extends TestCase
+{
+    private const DEFAULT_LANGUAGE_PLACEHOLDER = 'default_language';
+    private const SECOND_LANGUAGE_PLACEHOLDER = 'second_language';
+
+    private const DEFAULT_SHOP_PLACEHOLDER = 'default_shop';
+    private const SECOND_SHOP_PLACEHOLDER = 'second_shop';
+
+    /**
+     * @var int
+     */
+    private $defaultLanguageId;
+
+    /**
+     * @var int
+     */
+    private $secondLanguageId;
+
+    /**
+     * @var int
+     */
+    private $defaultShopId;
+
+    /**
+     * @var int
+     */
+    private $secondShopId;
+
+    protected function setUp(): void
```

Comment:
Shouldn't there be a teardown to remove what was inserted?

Consider doing the setup/teardown using setupBeforeClass and teardownBeforeClass to avoid doing this once per test case.

Reply:
I wanted to rely on setupBeforeClass/teardownBeforeClass But the problem is that this Intergation test environment uses `processIsolation=true` which seems to make `setupBeforeClass` called on EACH tests.. Basically setupBeforeClass acts almost like a regular setUp

This is really bothersome, because of this I wasn't able to make a proper independent initialization (which is why I need to always check if the data is present or not) nor can I do a cleaning because I can't detect which test is the last one

But I tried to clean as much as possible, so I still drop the content of `testable_object` table on each init, I could also drop them on tearDown and remove the extra shop and language, but it will result in creating/dropping each content on each test call which will make the whole test longer